migrate/20230920_postgres.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:postgres_server) do
+      column :id, :uuid, primary_key: true, default: nil
+      column :created_at, :timestamptz, null: false, default: Sequel.lit("now()")
+      column :updated_at, :timestamptz, null: false, default: Sequel.lit("now()")
+      column :project_id, :project, type: :uuid, null: false
```

Comment:
I think this is suspicious given the authorization model, although I can appreciate the convenience. Would a method do the job? technically more than one project can associate with a resource, though we don't use that feature for anything at present (it may be useful for transitioning from one set of tags and policies to another in a smooth way)

Reply:
@enescakir also raised this. Maybe this is good time to discuss N-to-N relationship of projects and resources.

I'm not entirely convinced on the benefit of being able to associate one resource to multiple projects at the cost of additional complexity. When we associate resources with multiple projects, we bring resource to the same level in the hierarchy as projects and accounts, and our model becomes more like graph than tree, which is more difficult to deal with.

On more practical side;
- Which project should be charged for a particular resource? For VMs we do projects.first, which is a bad hack IMO, but works as long as we don't introduce second project to the mix.
- With N-to-N association, deleting a project becomes a reference counting for each resource in it, which is doable, but why bother?
- Will we have no-access rules in our authorization model? I think we should, they are quite useful. However they would cause quite interesting problems with N-to-N relationship of projects and resources. Particularly, what do we do if a resource is in 2 projects and user defines access rule for one and no-access rule for the other.

Personally, I'd argue adding project_id to VMs as well. We can still keep N-to-N relationship between projects and accounts. Also users can still define access rules for individual resources.

Reply:
The only use case I have thought of for multi-project is transitioning from one security regime to another safely, being able to test it in phases. As it stands in industry today, this is an extremely narrow use case, just about everyone makes reorganizing the security apparatus for resources a high-stakes maneuver that cannot be tested, and everyone accepts it. This may be an acceptable thing to likewise inflict on people.

To enable two-phase opens non-trivial issues, like possibly having a secondary project that's insecure accessing a resource.

The era of developing this was quite a while ago, so perhaps our consideration was more favorable to experiment than tradition at that time. And since then, some things did come out, such as Enes's `Authorization` module. But in the current era it'd be more tempting to see if we could pare things back since we haven't made strides here in some time.

Reply:
This discussion also makes me think about the private subnets. What if customer wants to be able to connect 2 projects? Having 1 subnet listed under 2 projects make this much easier.

Reply:
We talked about this offline and decided to collect more data points to before making a decision on n-to-n relationships. For now, I kept the project_id. I'll also add a DB constraint to prevent potential unintentional disagreements between two models.